### PR TITLE
Enable transparent proxy

### PR DESCRIPTION
**Pull request by ishustava**
Changes proposed in this PR:
- This PR merges `feature-tproxy` into `master`. All changes in that branch has been individually reviewed and tested.
- Please see changelog

Note to reviewers: This is still waiting on the #486 changelog PR, but otherwise ready to review.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ executors:
       - image: docker.mirror.hashicorp.services/circleci/golang:1.14
     environment:
       TEST_RESULTS: /tmp/test-results # path to where test results are saved
-      CONSUL_VERSION: 1.9.0-rc1 # Consul's OSS version to use in tests
-      CONSUL_ENT_VERSION: 1.9.0+ent-rc1 # Consul's enterprise version to use in tests
+      CONSUL_VERSION: 1.9.4 # Consul's OSS version to use in tests
```
Comment (kschoche) [inline .circleci/config.yml:9]: So now I'm wondering if setting tproxy to default enabled will work without pointing this at the beta build of consul 1.10? @ishustava ?

Comment (ishustava) [reply .circleci/config.yml:9]: it should. The api changes are backward compatible and so the newer API package should work against the older agent.

Comment (kschoche) [reply .circleci/config.yml:9]: perfect, okay thanks for clarifying!

**Review by kschoche — APPROVED** [review]
Yeah!

**Review by lkysow — APPROVED** [review]
🚀 🎸  So awesome!

**Pull request merged**